packages/components/src/Lettera/Models.purs

```diff
@@ -18,22 +33,101 @@ isPreviewArticle :: FullArticle -> Boolean
 isPreviewArticle (PreviewArticle _) = true
 isPreviewArticle _ = false
 
-type ArticleStub =
-  { title    :: String
-  , uuid     :: String
-  , preamble :: String
+newtype LocalDateTime = LocalDateTime DateTime
+
+fromUTCTime :: DateTime -> Effect (Maybe LocalDateTime)
+fromUTCTime utcTime = do
+  let jsDate = JSDate.fromDateTime utcTime
+  offset <- (_ * -1.0) <$> JSDate.getTimezoneOffset jsDate
+  pure $ (LocalDateTime <$> adjust (Duration.Minutes offset) utcTime)
+
+type ArticleStubCommon =
+  ( title     :: String
+  , uuid      :: String
+  , preamble  :: String
   , listImage :: Maybe Image
-  , tags :: Array String
-  , premium :: Boolean
+  , tags      :: Array String
+  , premium   :: Boolean
+  )
+
+type JSArticleStub =
+  { publishingTime :: String
+  | ArticleStubCommon
   }
 
-type Article =
-  { title     :: String
+type ArticleStub =
+  { publishingTime :: LocalDateTime
+  | ArticleStubCommon
+  }
+
+type ArticleCommon =
+  ( title     :: String
   , body      :: Array BodyElementJS
   , mainImage :: Maybe Image
   , tags      :: Array String
+  , uuid      :: String
+  , authors   :: Array Author
+  , premium   :: Boolean
+  )
+
+type JSArticle =
+  { publishingTime :: String
+  , updateTime     :: Maybe String
+  | ArticleCommon
   }
 
+type Article =
+  { publishingTime :: LocalDateTime
+  , updateTime     :: Maybe LocalDateTime
+  | ArticleCommon
+  }
+
+type Author =
+  { byline :: String
+  , image  :: Maybe String
+  }
+
+parseArticleWith :: forall a b. ReadForeign b => (b -> Effect (Maybe a)) -> Json -> Effect (Either String a)
+parseArticleWith parseFn articleResponse = do
+  case JSON.readJSON $ stringify articleResponse of
+    Right jsArticle -> do
+      maybeParsedArticle <- parseFn jsArticle
+      case maybeParsedArticle of
+        Just article -> pure $ Right article
+        _            -> do
+          Console.warn $ "Could not parse API article into article"
+          pure $ Left "Parsing error"
+    Left err -> do
+      let parsingErrors = joinWith " " $ fromFoldable $ map renderForeignError err
+      -- TODO: Sentry and whatnot
+      Console.warn $ "Could not parse article JSON, errors: " <> parsingErrors
+      pure $ Left $ "Parsing error: " <> parsingErrors
+
+parseArticle :: Json -> Effect (Either String Article)
+parseArticle = parseArticleWith fromJSArticle
+
+parseArticleStub :: Json -> Effect (Either String ArticleStub)
+parseArticleStub = parseArticleWith fromJSArticleStub
+
+parseDateTime :: String -> Maybe DateTime
+parseDateTime = hush <<< unformat dateTimeFormatter
+
+fromJSArticleStub :: JSArticleStub -> Effect (Maybe ArticleStub)
+fromJSArticleStub jsStub@{ publishingTime } = runMaybeT do
```

Comment:
This function and `fromJsArticle` are bound to fail if the timestamps cannot be parsed. I’m not sure this would be the best way to go at it.

What would happen if there was a change in the format of the timestamps? Can we guarantee that it won’t change unexpectedly for the client?

If such a thing happened, no article would be displayed until the problem is fixed.

Reply:
Good point. Will think about this

Reply:
Decided to turn this into `Maybe LocalDateTime` as defaulting it to some nonsense value on parsing error is pointless.